models/classes/LtiOutcome/LtiOutcomeXmlFactory.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA
+ */
+
+namespace oat\taoLti\models\classes\LtiOutcome;
+
+use oat\oatbox\service\ConfigurableService;
+
+class LtiOutcomeXmlFactory extends ConfigurableService
+{
+
+    private const REPLACE_RESULT_REQUEST = 'replaceResultRequest';
+    private const OUTCOME_DEFAULT_LANG = 'en-us';
+
+    public function build(
```

Comment:
Instead of having to delegate the client to know the `operation`, the factory method could be more explicit, like `buildReplaceResultRequest()`. Wdyt? If not, we might have to validate this parameter, what I think can be avoided.

Reply:
done